### DISCUSSION TITLE
Text-Field property depending on value of a field

### DISCUSSION THREAD

**Discussion by sylvain258** [maplibre/maplibre-gl-js#4516] [Q&A]
Hello,
I'm looking for a symbology depending on the value of a field. I have a field with several distinct values and for each case, I would like to change the case, size and colors of the fonts according to these values. For example, I have a "color" field with 4 distinct values. For the value "Red", I'd like the wording to be in uppercase and red in color; for the value "Blue", in lowercase and blue in color and so on.
I haven't found my request on other forums.

Thank you for your help,

Sylvain Becquaert

**Comment by HarelM** [comment]
A visual representation (mocked) would help better understand what exactly you want to achieve.
But basically you can define a few layers and filter out this layer based on data from the source. each layer can have different style so that you can achieve different map look based on different data.
You can complicate the filter or data expressions to achieve more complicated scenarios that might not need to duplicate layers.
See here:
https://maplibre.org/maplibre-style-spec/expressions/

**Comment by sylvain258** [comment]
Hello,
![Capture d’écran 2024-08-07 à 11 07 26](https://github.com/user-attachments/assets/66ca66f6-3caa-47e0-a337-c7f546b37fff)
That is what I want to do. I studied Maplibre expressions but there isn't my case.
Thank you for your help

**Reply by HarelM** [reply]
It is still not clear what you want to achieve...

**Reply by sylvain258** [reply]
I am going to generate style with Mapbox Studio. Thanks :-)
